\appendix
\appendixpage*
\addappheadtotoc

\chapter{Basis matrix \texorpdfstring{$W$}{W} for the six survival-associated metagenes}
\label{app:sigs-w-matrix}
\input{analysis/biosurv/reports/18_SIS_diag_dsd_final/basis_table.tex}

\chapter{\texorpdfstring{\acrshort{MSIGDB}}{MSIGDB} signatures correlated with axis A1}
\label{app:sigs-msigdb-corrs-axis1}
\begin{longtable}[!htbp]{ l r@{.}l }
\caption[\texorpdfstring{\acrshort{MSIGDB}}{MSIGDB} signatures correlated with axis A1]{\acrshort{MSIGDB} signatures substantially correlated with activity of the prognostic axis A1.}

\hline \multicolumn{1}{l}{MSigDB set} & \multicolumn{2}{r}{A1 correlation} \\ \hline
\endfirsthead

\multicolumn{3}{l}
{{\bfseries \tablename\ \thetable{} -- continued from previous page}} \\
\hline \multicolumn{1}{l}{MSigDB set} & \multicolumn{2}{r}{A1 correlation} \\ \hline
\endhead

\hline \multicolumn{3}{r}{{Continued on next page}} \\
\endfoot

\hline \hline
\endlastfoot

c5.M\_PHASE / c5.MITOSIS / \\
\qquad c5.M\_PHASE\_OF\_MITOTIC\_CELL\_CYCLE & $0$ & $689$ \\
c5.REGULATION\_OF\_MITOSIS & $0$ & $682$ \\
c4.GNF2\_RFC3 / c4.GNF2\_RFC4 / c4.GNF2\_SMC2L1 / \\
\qquad c4.GNF2\_CKS1B / c4.GNF2\_CKS2 / c4.GNF2\_TTK & $0$ & $664$ \\
c5.CELL\_CYCLE\_PROCESS / \\
\qquad c5.MITOTIC\_CELL\_CYCLE / \\
\qquad c5.CELL\_CYCLE\_PHASE & $0$ & $653$ \\
c5.SPINDLE & $0$ & $644$ \\
c4.MORF\_BUB1B & $0$ & $631$ \\
c6.CSR\_LATE\_UP.V1\_SIGNED & $0$ & $630$ \\
c5.SPINDLE\_POLE & $0$ & $628$ \\
c2.PID\_PLK1\_PATHWAY & $0$ & $626$ \\
c5.ORGANELLE\_PART / \\
\qquad c5.INTRACELLULAR\_ORGANELLE\_PART & $0$ & $624$ \\
c2.REACTOME\_CELL\_CYCLE / \\
\qquad c2.REACTOME\_CELL\_CYCLE\_MITOTIC & $0$ & $622$ \\
c2.REACTOME\_CYCLIN\_A\_B1\_ASSOCIATED\_ \\
\qquad EVENTS\_DURING\_G2\_M\_TRANSITION & $0$ & $604$ \\
c2.REACTOME\_MITOTIC\_PROMETAPHASE & $0$ & $596$ \\
c2.KEGG\_CELL\_CYCLE & $0$ & $588$ \\
c5.CHROMOSOME\_SEGREGATION & $0$ & $588$ \\
c4.MORF\_FEN1 & $0$ & $586$ \\
c2.REACTOME\_G1\_S\_SPECIFIC\_TRANSCRIPTION & $0$ & $585$ \\
c2.REACTOME\_ACTIVATION\_OF\_THE\_ \\
\qquad PRE\_REPLICATIVE\_COMPLEX / \\
\qquad c2.REACTOME\_ACTIVATION\_OF\_ATR\_ \\ 
\qquad IN\_RESPONSE\_TO\_REPLICATION\_STRESS / \\ 
\qquad c2.REACTOME\_G2\_M\_CHECKPOINTS & $0$ & $583$ \\
c2.REACTOME\_E2F\_ENABLED\_INHIBITION\_OF\_ \\
\qquad PRE\_REPLICATION\_COMPLEX\_FORMATION & $0$ & $581$ \\
c2.REACTOME\_E2F\_MEDIATED\_REGULATION\_OF\_ \\
\qquad DNA\_REPLICATION & $0$ & $577$ \\
c5.CELL\_CYCLE\_GO\_0007049 & $0$ & $576$ \\
c2.REACTOME\_KINESINS & $0$ & $575$ \\
c3.V\$ELK1\_02 & $0$ & $574$ \\
c5.SPINDLE\_MICROTUBULE & $0$ & $573$ \\
c5.MITOTIC\_CELL\_CYCLE\_CHECKPOINT & $0$ & $569$ \\
c2.REACTOME\_CELL\_CYCLE\_CHECKPOINTS / \\
\qquad c2.REACTOME\_G1\_S\_TRANSITION / \\ 
\qquad c2.REACTOME\_SYNTHESIS\_OF\_DNA / \\
\qquad c2.REACTOME\_MITOTIC\_G1\_G1\_S\_PHASES / \\ 
\qquad c2.REACTOME\_MITOTIC\_M\_M\_G1\_PHASES / \\
\qquad c2.REACTOME\_DNA\_REPLICATION / \\
\qquad c2.REACTOME\_S\_PHASE & $0$ & $566$ \\
c4.MORF\_ESPL1 & $0$ & $566$ \\
c4.MORF\_BUB1 & $0$ & $565$ \\
c4.MORF\_BUB3/c4.MORF\_RAD23A & $0$ & $563$ \\
c5.CONDENSED\_CHROMOSOME & $0$ & $562$ \\
c4.MORF\_RFC4/c4.MORF\_RRM1 & $0$ & $561$ \\
c2.BIOCARTA\_G2\_PATHWAY & $0$ & $559$ \\
c3.SCGGAAGY\_V\$ELK1\_02 & $0$ & $558$ \\
c2.PID\_AURORA\_A\_PATHWAY & $0$ & $556$ \\
c5.MITOTIC\_SISTER\_CHROMATID\_SEGREGATION / \\
\qquad c5.SISTER\_CHROMATID\_SEGREGATION & $0$ & $555$ \\
c4.MORF\_UNG & $0$ & $554$ \\
c2.PID\_FOXM1PATHWAY & $0$ & $551$ \\
c4.MORF\_GSPT1 & $0$ & $550$ \\
c2.REACTOME\_METABOLISM\_OF\_NUCLEOTIDES & $0$ & $550$ \\
c2.PID\_ATR\_PATHWAY & $0$ & $547$ \\
c2.BIOCARTA\_MCM\_PATHWAY & $0$ & $546$ \\
c4.MORF\_CCNF & $0$ & $544$ \\
c5.CELL\_CYCLE\_CHECKPOINT\_GO\_0000075 & $0$ & $543$ \\
c5.MITOTIC\_SPINDLE\_ORGANIZATION\_AND\_ \\
\qquad BIOGENESIS / \\
\qquad c5.SPINDLE\_ORGANIZATION\_AND\_BIOGENESIS & $0$ & $542$ \\
c4.MORF\_EI24 & $0$ & $538$ \\
c5.DOUBLE\_STRAND\_BREAK\_REPAIR & $0$ & $537$ \\
c4.GNF2\_PA2G4/c4.GNF2\_RAN & $0$ & $531$ \\
c2.REACTOME\_G2\_M\_DNA\_DAMAGE\_CHECKPOINT & $0$ & $531$ \\
c2.KEGG\_PYRIMIDINE\_METABOLISM & $0$ & $531$ \\
c4.MORF\_GMPS & $0$ & $528$ \\
c4.MORF\_PRKDC & $0$ & $528$ \\
c2.PID\_BARD1PATHWAY & $0$ & $528$ \\
c4.GNF2\_MCM5 & $0$ & $525$ \\
c4.MORF\_DNMT1 & $0$ & $524$ \\
c2.REACTOME\_POL\_SWITCHING & $0$ & $523$ \\
c4.GNF2\_MSH2 & $0$ & $521$ \\
c4.MORF\_CSNK2B & $0$ & $520$ \\
c2.PID\_AURORA\_B\_PATHWAY & $0$ & $520$ \\
c2.REACTOME\_DESTABILIZATION\_OF\_MRNA\_BY\_KSRP & $0$ & $517$ \\
c5.DNA\_METABOLIC\_PROCESS & $0$ & $517$ \\
c4.MORF\_PTPN11 & $0$ & $516$ \\
c5.REGULATION\_OF\_MITOTIC\_CELL\_CYCLE & $0$ & $516$ \\
c5.RESPONSE\_TO\_ENDOGENOUS\_STIMULUS / \\
\qquad c5.RESPONSE\_TO\_DNA\_DAMAGE\_STIMULUS & $0$ & $515$ \\
c5.CHROMOSOMEPERICENTRIC\_REGION / \\
\qquad c5.KINETOCHORE & $0$ & $514$ \\
c6.MTOR\_UP.V1\_SIGNED & $0$ & $512$ \\
c2.REACTOME\_APOPTOSIS & $0$ & $510$ \\
c4.MORF\_PPP1CC & $0$ & $509$ \\
c5.PORE\_COMPLEX/c5.NUCLEAR\_PORE & $0$ & $508$ \\
c5.DNA\_REPAIR & $0$ & $506$ \\
c2.REACTOME\_CHROMOSOME\_MAINTENANCE / \\
\qquad c2.REACTOME\_TELOMERE\_MAINTENANCE & $0$ & $506$ \\
c5.MACROMOLECULAR\_COMPLEX / \\
\qquad c5.PROTEIN\_COMPLEX & $0$ & $506$ \\
c4.MORF\_XRCC5/c4.MORF\_GNB1 & $0$ & $504$ \\
c5.INTERPHASE / \\
\qquad c5.INTERPHASE\_OF\_MITOTIC\_CELL\_CYCLE & $0$ & $503$ \\
c5.NON\_MEMBRANE\_BOUND\_ORGANELLE / \\
\qquad c5.INTRACELLULAR\_NON\_ \\
\qquad MEMBRANE\_BOUND\_ORGANELLE & $0$ & $503$ \\
c6.GCNP\_SHH\_UP\_EARLY.V1\_SIGNED & $0$ & $503$ \\
c2.BIOCARTA\_RANMS\_PATHWAY & $0$ & $502$ \\
c2.KEGG\_DNA\_REPLICATION / \\
\qquad c2.REACTOME\_DNA\_STRAND\_ELONGATION & $0$ & $502$ \\
c4.MORF\_SOD1 & $0$ & $502$ \\
c5.NUCLEAR\_MEMBRANE / \\ 
\qquad c5.NUCLEAR\_MEMBRANE\_PART & $0$ & $501$ \\
c4.MORF\_HDAC1 & $0$ & $501$ \\
c2.REACTOME\_HIV\_LIFE\_CYCLE / \\
\qquad c2.REACTOME\_LATE\_PHASE\_OF\_HIV\_LIFE\_CYCLE & $0$ & $500$ \\
c5.CHROMOSOMAL\_PART/c5.CHROMOSOME & $0$ & $500$ \\
c5.PHOSPHORIC\_DIESTER\_HYDROLASE\_ACTIVITY & $-0$ & $502$ \\
c3.CTGCAGY\_UNKNOWN & $-0$ & $505$ \\
c3.V\$OCT1\_01 & $-0$ & $509$ \\
c3.V\$GATA\_Q6 & $-0$ & $515$ \\
c5.CELL\_SURFACE\_RECEPTOR\_LINKED\_ \\
\qquad SIGNAL\_TRANSDUCTION\_GO\_0007166 & $-0$ & $518$ \\
c4.GNF2\_MAPT & $-0$ & $526$ \\
c3.V\$OCT1\_04 & $-0$ & $531$ \\
c2.REACTOME\_G\_ALPHA\_S\_SIGNALLING\_EVENTS & $-0$ & $539$ \\
c3.V\$OCT\_C & $-0$ & $544$ \\
\end{longtable}



\chapter{\texorpdfstring{\acrshort{MSIGDB}}{MSIGDB} signatures correlated with axis A2}
\label{app:sigs-msigdb-corrs-axis2}
\begin{longtable}[!htbp]{ l r@{.}l }
\caption[\texorpdfstring{\acrshort{MSIGDB}}{MSIGDB} signatures correlated with axis A2]{\acrshort{MSIGDB} signatures substantially correlated with activity of the prognostic axis A2.}

\hline \multicolumn{1}{l}{MSigDB set} & \multicolumn{2}{r}{A2 correlation} \\ \hline
\endfirsthead

\multicolumn{3}{l}
{{\bfseries \tablename\ \thetable{} -- continued from previous page}} \\
\hline \multicolumn{1}{l}{MSigDB set} & \multicolumn{2}{r}{A2 correlation} \\ \hline
\endhead

\hline \multicolumn{3}{r}{{Continued on next page}} \\
\endfoot

\hline \hline
\endlastfoot

c2.PID\_INTEGRIN1\_PATHWAY                          & $0$ & $654$       \\
c2.PID\_INTEGRIN3\_PATHWAY                          & $0$ & $637$       \\
c2.PID\_UPA\_UPAR\_PATHWAY                          & $0$ & $597$       \\
c4.GNF2\_PTX3                                       & $0$ & $593$       \\
c2.KEGG\_ECM\_RECEPTOR\_INTERACTION                 & $0$ & $582$       \\
c2.PID\_INTEGRIN5\_PATHWAY                          & $0$ & $577$       \\
c4.GNF2\_MMP1                                       & $0$ & $575$       \\
c2.REACTOME\_EXTRACELLULAR\_MATRIX\_                                    \\
\qquad ORGANIZATION /                                                   \\
\qquad c2.REACTOME\_COLLAGEN\_FORMATION             & $0$ & $572$       \\
c5.AXON\_GUIDANCE                                   & $0$ & $571$       \\
c2.KEGG\_FOCAL\_ADHESION                            & $0$ & $567$       \\
c2.PID\_SYNDECAN\_1\_PATHWAY                        & $0$ & $552$       \\
c2.REACTOME\_CELL\_EXTRACELLULAR\_                                      \\
\qquad MATRIX\_INTERACTIONS                         & $0$ & $538$       \\
c2.PID\_INTEGRIN\_CS\_PATHWAY                       & $0$ & $536$       \\
c5.TISSUE\_DEVELOPMENT                              & $0$ & $536$       \\
c5.COLLAGEN                                         & $0$ & $531$       \\
c6.CORDENONSI\_YAP\_CONSERVED\_SIGNATURE            & $0$ & $526$       \\
c6.LEF1\_UP.V1\_SIGNED                              & $0$ & $519$       \\
c2.REACTOME\_INTEGRIN\_CELL\_SURFACE\_                                  \\
\qquad INTERACTIONS                                 & $0$ & $518$       \\
c5.AXONOGENESIS /                                                       \\
\qquad c5.CELLULAR\_MORPHOGENESIS\_                                     \\
\qquad DURING\_DIFFERENTIATION                      & $0$ & $515$       \\
c6.STK33\_NOMO\_SIGNED                              & $0$ & $507$       \\
c7.GSE17721\_CTRL\_VS\_CPG\_12H\_BMDM\_SIGNED       & $-0$ & $508$      \\
c7.GSE1460\_INTRATHYMIC\_T\_PROGENITOR\_VS\_                            \\
\qquad THYMIC\_STROMAL\_CELL\_SIGNED                & $-0$ & $508$      \\ 
\end{longtable}


\chapter{Approximate calculation of \texorpdfstring{\acrshort{PARSE}}{PARSE} scores}
\label{app:sigs-parse-approx}
Exact calculation of \gls{PARSE} score requires the solution of a number of \gls{NNLS} problems, which complicates application.  The \gls{NNLS} solutions can be approximated with conventional least squares solutions, ultimately transforming the calculation of an approximate \gls{PARSE} score into a simple weighted sum of gene expression measurements.

Recall that \gls{NMF} finds factorizations of the form $A = W H$, with all elements of $A$, $W$, and $H$, being non-negative.  In the reverse problem of \gls{PARSE} calculation, $A$ and $\widehat{W}$ are supplied, and $H$ is to be estimated.  I propose an approximation that removes the requirement that $H$ be non-negative, $H \approx \widehat{W}^+ A$, where $\widehat{W}^+$ is the Moore-Penrose pseudoinverse of $\widehat{W}$.  By combining this approximation with the linear combination of metagene coefficients that forms the \gls{PARSE} score, we can approximate \gls{PARSE} as a simple weighted sum of gene expression measurements:
\begin{align}
  P &= L H \\
    &\approx L \widehat{W}^+ A \\
    &= k A
\end{align}
where $P$ is the vector of \gls{PARSE} score values, $L$ is the metagene loadings for the \gls{PARSE} score, $L = \irow{1.354&-1.548&0&0&-1.354&1.548}$, and $k$ is a row vector of gene loadings for calculation of an approximate \gls{PARSE} score.  Approximation of $P$ by $k A$ appears excellent; when tested on \gls{APGI} gene expression measurements, the approximation closely matched the more laborious exact \gls{NNLS} solution (\fref{fig:app-parse-approx-matching}).

\begin{figure}
\centering
\includegraphics[width=.7\linewidth]{analysis/biosurv/reports/18_SIS_diag_dsd_final/figure/approx-calc-1}
\caption[Performance of the \texorpdfstring{\acrshort{PARSE}}{PARSE} score approximation]{The linear \acrshort{PARSE} score approximation $P \approx k A$ closely matches the exact version calculated using \gls{NNLS}, when evaluated on \gls{APGI} \gls{GEX} data.}\label{fig:app-parse-approx-matching}
\end{figure}

To use the approximation in practice, perform the following steps:
\begin{enumerate}
  \item Prepare a gene $\times$ sample matrix of linear expression estimates $A$, in which values for each row (gene) have been scaled to encompass the range $0$ to $1$.
  \item Subset $A$ to only the genes present in the $k$ vector (\tref{tab:sigs-approx-parse-loadings}), and arrange rows of $A$ so that they exactly match the order of $k$.  If genes present in $k$ are missing from $A$, insert all-zero rows for these genes into $A$.
  \item Calculate approximate \gls{PARSE} scores $P$ as $P = k A$.  This is equivalent to, for each column (sample) of $A$, multiplying each entry of the column of $A$ with the corresponding entry of $k$, and summing the results.
\end{enumerate}

The loading vector for the calculation of approximate \gls{PARSE} score, $k^T$, is given in \tref{tab:sigs-approx-parse-loadings}.

\begin{longtable}[!htbp]{ l r@{.}l @{\hspace{60pt}} l r@{.}l }
\caption[Loading vector for the approximate \texorpdfstring{\acrshort{PARSE}}{PARSE} score]{Loading vector for the approximate \acrshort{PARSE} score.  For brevity and to assist interpretation, this has been split by sign into two columns, but in use both columns should be combined to form a single row vector $k$.}
\label{tab:sigs-approx-parse-loadings}

\hline Gene symbol & \multicolumn{2}{l}{Loading} & Gene symbol & \multicolumn{2}{r}{Loading} \\ \hline
\endfirsthead

\multicolumn{6}{l}{{\bfseries \tablename\ \thetable{} -- continued from previous page}} \\
\hline Gene symbol & \multicolumn{2}{l}{Loading} & Gene symbol & \multicolumn{2}{r}{Loading} \\ \hline
\endhead

\hline \multicolumn{6}{r}{{Continued on next page}} \\
\endfoot

\hline \hline
\endlastfoot

FEM1B & $0$ & $04785$ & GAB2 & $-0$ & $03742$ \\
NCAPG & $0$ & $04487$ & FRZB & $-0$ & $03715$ \\
ANLN & $0$ & $04364$ & MIR99AHG & $-0$ & $03712$ \\
COL12A1 & $0$ & $04098$ & RAP1GAP & $-0$ & $03483$ \\
LDHA & $0$ & $04004$ & NFIA & $-0$ & $03387$ \\
E2F7 & $0$ & $03923$ & TCTA & $-0$ & $03326$ \\
SPHK1 & $0$ & $03861$ & ELMOD3 & $-0$ & $03300$ \\
CEP55 & $0$ & $03755$ & SOBP & $-0$ & $03269$ \\
CHEK1 & $0$ & $03669$ & GIMAP2 & $-0$ & $03176$ \\
TMEM26 & $0$ & $03659$ & STAT5B & $-0$ & $03172$ \\
CKAP2L & $0$ & $03545$ & UFC1 & $-0$ & $03123$ \\
DCBLD2 & $0$ & $03351$ & BOC & $-0$ & $03047$ \\
PHLDA1 & $0$ & $03330$ & P2RY8 & $-0$ & $03043$ \\
KANK4 & $0$ & $03261$ & RNF103 & $-0$ & $03019$ \\
TGFBI & $0$ & $03259$ & KIAA0513 & $-0$ & $02989$ \\
PLAU & $0$ & $03213$ & SGSM1 & $-0$ & $02933$ \\
COL5A3 & $0$ & $03177$ & TOR2A & $-0$ & $02926$ \\
CCNB1 & $0$ & $03071$ & PPY & $-0$ & $02787$ \\
SPOCK1 & $0$ & $03046$ & SH3GL1 & $-0$ & $02784$ \\
ENO2 & $0$ & $02998$ & RPA2 & $-0$ & $02756$ \\
CAV1 & $0$ & $02989$ & SELENBP1 & $-0$ & $02707$ \\
KIF20A & $0$ & $02967$ & TRIM2 & $-0$ & $02689$ \\
RACGAP1 & $0$ & $02957$ & TCEA3 & $-0$ & $02679$ \\
PPAPDC1A & $0$ & $02867$ & HIPK2 & $-0$ & $02620$ \\
RBMS2 & $0$ & $02834$ & CAPN6 & $-0$ & $02615$ \\
RHOF & $0$ & $02828$ & ARHGAP24 & $-0$ & $02524$ \\
CDA & $0$ & $02792$ & TSTD1 & $-0$ & $02503$ \\
NCAPD2 & $0$ & $02756$ & ALDH5A1 & $-0$ & $02452$ \\
MCM4 & $0$ & $02708$ & BCKDK & $-0$ & $02452$ \\
LOX & $0$ & $02695$ & GPC3 & $-0$ & $02419$ \\
PTGES & $0$ & $02681$ & EPHX2 & $-0$ & $02392$ \\
FER & $0$ & $02675$ & DCAF8 & $-0$ & $02374$ \\
EYA3 & $0$ & $02671$ & PPM1H & $-0$ & $02311$ \\
IL20RB & $0$ & $02671$ & PRDM16 & $-0$ & $02289$ \\
GATC & $0$ & $02661$ & MC1R & $-0$ & $02281$ \\
KLHL5 & $0$ & $02641$ & PEX11B & $-0$ & $02280$ \\
ARL4C & $0$ & $02609$ & SMOX & $-0$ & $02258$ \\
ATAD2 & $0$ & $02602$ & LMO3 & $-0$ & $02246$ \\
TPX2 & $0$ & $02590$ & RPIA & $-0$ & $02226$ \\
FGD6 & $0$ & $02545$ & POU2AF1 & $-0$ & $02222$ \\
PRC1 & $0$ & $02492$ & ST3GAL2 & $-0$ & $02187$ \\
MCM10 & $0$ & $02451$ & ZSCAN32 & $-0$ & $02184$ \\
BIRC5 & $0$ & $02419$ & ZFPM1 & $-0$ & $02180$ \\
ZBED2 & $0$ & $02396$ & BCL11B & $-0$ & $02161$ \\
KNTC1 & $0$ & $02375$ & C9orf152 & $-0$ & $02152$ \\
NUP155 & $0$ & $02330$ & SLC40A1 & $-0$ & $02146$ \\
TNFRSF6B & $0$ & $02308$ & CADPS2 & $-0$ & $02136$ \\
HJURP & $0$ & $02296$ & PHOSPHO2 & $-0$ & $02129$ \\
PXDN & $0$ & $02281$ & ST6GAL1 & $-0$ & $02118$ \\
COLGALT1 & $0$ & $02272$ & PLAC9 & $-0$ & $02093$ \\
PLOD2 & $0$ & $02261$ & EIF2AK3 & $-0$ & $02073$ \\
TWIST1 & $0$ & $02246$ & IFT140 & $-0$ & $02068$ \\
RALGAPB & $0$ & $02214$ & CHN2 & $-0$ & $02051$ \\
FSCN1 & $0$ & $02159$ & ZNF658 & $-0$ & $01988$ \\
SPOCD1 & $0$ & $02117$ & MEOX1 & $-0$ & $01961$ \\
SERPINH1 & $0$ & $02086$ & FAM134B & $-0$ & $01945$ \\
GAPDH & $0$ & $02073$ & THSD7B & $-0$ & $01931$ \\
DSG3 & $0$ & $02070$ & TRAPPC2 & $-0$ & $01920$ \\
MELK & $0$ & $02067$ & ADH1A & $-0$ & $01845$ \\
DCUN1D5 & $0$ & $02056$ & LINC01184 & $-0$ & $01837$ \\
TUBA1C & $0$ & $02053$ & SLC12A2 & $-0$ & $01821$ \\
CST6 & $0$ & $02032$ & MRAP2 & $-0$ & $01810$ \\
GABPB1 & $0$ & $01929$ & RASL11B & $-0$ & $01808$ \\
KRT7 & $0$ & $01916$ & RERGL & $-0$ & $01801$ \\
DENND1A & $0$ & $01898$ & PREP & $-0$ & $01799$ \\
AURKB & $0$ & $01869$ & TMTC4 & $-0$ & $01797$ \\
PRR11 & $0$ & $01859$ & TMED1 & $-0$ & $01796$ \\
RFC5 & $0$ & $01848$ & TLE4 & $-0$ & $01794$ \\
SLC16A3 & $0$ & $01842$ & CAMK1G & $-0$ & $01790$ \\
SUGCT & $0$ & $01833$ & GATA6 & $-0$ & $01780$ \\
SCYL2 & $0$ & $01826$ & CCR7 & $-0$ & $01775$ \\
KRT6A & $0$ & $01795$ & SCGB2A1 & $-0$ & $01773$ \\
P4HA2 & $0$ & $01770$ & CCL19 & $-0$ & $01715$ \\
PROSER2 & $0$ & $01761$ & PCF11 & $-0$ & $01710$ \\
PTPN21 & $0$ & $01723$ & FAM189A2 & $-0$ & $01692$ \\
PYGL & $0$ & $01714$ & MCOLN2 & $-0$ & $01684$ \\
GINS2 & $0$ & $01713$ & PLEKHS1 & $-0$ & $01672$ \\
PGBD3 & $0$ & $01700$ & PRMT7 & $-0$ & $01665$ \\
COL7A1 & $0$ & $01688$ & AXIN2 & $-0$ & $01658$ \\
LETM2 & $0$ & $01687$ & TOM1 & $-0$ & $01640$ \\
PDLIM7 & $0$ & $01678$ & RERE & $-0$ & $01635$ \\
KRT17 & $0$ & $01644$ & A4GNT & $-0$ & $01632$ \\
ERRFI1 & $0$ & $01597$ & CDK12 & $-0$ & $01624$ \\
ASPM & $0$ & $01593$ & CNNM1 & $-0$ & $01611$ \\
C1QTNF6 & $0$ & $01572$ & HSPB6 & $-0$ & $01586$ \\
DERA & $0$ & $01568$ & LCNL1 & $-0$ & $01571$ \\
MIF & $0$ & $01560$ & MTRNR2L1 & $-0$ & $01563$ \\
C5orf46 & $0$ & $01559$ & DYNC2H1 & $-0$ & $01537$ \\
EMP3 & $0$ & $01550$ & NPM1 & $-0$ & $01520$ \\
CDK2 & $0$ & $01546$ & CARHSP1 & $-0$ & $01515$ \\
POC1A & $0$ & $01507$ & RGS5 & $-0$ & $01505$ \\
FST & $0$ & $01504$ & CLEC3B & $-0$ & $01500$ \\
KCTD10 & $0$ & $01501$ & NR0B2 & $-0$ & $01468$ \\
SULF2 & $0$ & $01494$ & ARSD & $-0$ & $01466$ \\
CCDC88A & $0$ & $01480$ & GNPAT & $-0$ & $01458$ \\
KIF14 & $0$ & $01477$ & MARS2 & $-0$ & $01442$ \\
DSG2 & $0$ & $01463$ & KCTD5 & $-0$ & $01440$ \\
UHRF2 & $0$ & $01445$ & MRPL24 & $-0$ & $01395$ \\
ZNF185 & $0$ & $01435$ & ABLIM1 & $-0$ & $01392$ \\
SLC2A1 & $0$ & $01424$ & ITPKB & $-0$ & $01390$ \\
KIF2C & $0$ & $01417$ & FHDC1 & $-0$ & $01380$ \\
FLRT3 & $0$ & $01416$ & C2orf70 & $-0$ & $01360$ \\
CNIH3 & $0$ & $01413$ & RAVER2 & $-0$ & $01352$ \\
ITGA5 & $0$ & $01407$ & AKR1A1 & $-0$ & $01321$ \\
DNAJC9 & $0$ & $01385$ & CACHD1 & $-0$ & $01313$ \\
ANGPTL4 & $0$ & $01365$ & ACYP2 & $-0$ & $01298$ \\
KIAA1549L & $0$ & $01354$ & CTSL & $-0$ & $01263$ \\
PPP1R14B & $0$ & $01352$ & TM9SF3 & $-0$ & $01255$ \\
PAX8 & $0$ & $01350$ & PP7080 & $-0$ & $01242$ \\
FAM91A1 & $0$ & $01341$ & IGLL3P & $-0$ & $01241$ \\
EDIL3 & $0$ & $01326$ & ST6GALNAC1 & $-0$ & $01232$ \\
RAB31 & $0$ & $01316$ & VPS35 & $-0$ & $01219$ \\
P2RY2 & $0$ & $01288$ & TAF5L & $-0$ & $01213$ \\
CDC45 & $0$ & $01256$ & STK39 & $-0$ & $01196$ \\
SPIN4 & $0$ & $01254$ & NFIX & $-0$ & $01186$ \\
APCDD1 & $0$ & $01244$ & TNFRSF17 & $-0$ & $01180$ \\
ABHD5 & $0$ & $01227$ & PBXIP1 & $-0$ & $01174$ \\
ANKLE2 & $0$ & $01205$ & PLIN2 & $-0$ & $01174$ \\
FAM83A & $0$ & $01202$ & GOLM1 & $-0$ & $01171$ \\
KYNU & $0$ & $01181$ & SEPW1 & $-0$ & $01161$ \\
ANGPTL2 & $0$ & $01178$ & FYN & $-0$ & $01133$ \\
B3GALTL & $0$ & $01113$ & CA8 & $-0$ & $01129$ \\
MME & $0$ & $01102$ & CSNK1D & $-0$ & $01128$ \\
FAH & $0$ & $01035$ & SLC30A3 & $-0$ & $01126$ \\
NEURL2 & $0$ & $01012$ & SEC23IP & $-0$ & $01125$ \\
CTSV & $0$ & $00987$ & RFK & $-0$ & $01090$ \\
PGAM5 & $0$ & $00973$ & SDIM1 & $-0$ & $01083$ \\
ATL3 & $0$ & $00972$ & ARFGAP3 & $-0$ & $01070$ \\
CD70 & $0$ & $00954$ & CYP2S1 & $-0$ & $01044$ \\
CHAF1B & $0$ & $00920$ & TARBP2 & $-0$ & $01019$ \\
PIGL & $0$ & $00833$ & SERTAD2 & $-0$ & $00995$ \\
PAX8-AS1 & $0$ & $00830$ & IL33 & $-0$ & $00991$ \\
LMTK2 & $0$ & $00804$ & FAM120AOS & $-0$ & $00980$ \\
ACKR3 & $0$ & $00802$ & SYNE2 & $-0$ & $00968$ \\
KRT6C & $0$ & $00798$ & COX4I2 & $-0$ & $00943$ \\
PRKCDBP & $0$ & $00755$ & ANKRD22 & $-0$ & $00941$ \\
DPY19L1 & $0$ & $00749$ & COLGALT2 & $-0$ & $00903$ \\
NACC2 & $0$ & $00733$ & FBXW8 & $-0$ & $00891$ \\
POLA2 & $0$ & $00692$ & MARCKSL1 & $-0$ & $00884$ \\
DKK1 & $0$ & $00649$ & BTN3A1 & $-0$ & $00868$ \\
FBXO22 & $0$ & $00649$ & C1orf56 & $-0$ & $00865$ \\
USP30 & $0$ & $00629$ & PCDH20 & $-0$ & $00861$ \\
APCS & $0$ & $00602$ & EXOSC8 & $-0$ & $00850$ \\
BBS2 & $0$ & $00587$ & AMOT & $-0$ & $00825$ \\
TRERF1 & $0$ & $00581$ & WNT2B & $-0$ & $00812$ \\
GPR176 & $0$ & $00563$ & SLAMF9 & $-0$ & $00761$ \\
FGG & $0$ & $00548$ & PCOLCE2 & $-0$ & $00752$ \\
AKIP1 & $0$ & $00545$ & ZSCAN16 & $-0$ & $00720$ \\
IDH2 & $0$ & $00528$ & CIDECP & $-0$ & $00684$ \\
PFKFB4 & $0$ & $00525$ & BAMBI & $-0$ & $00680$ \\
ANKRD37 & $0$ & $00474$ & IL1R2 & $-0$ & $00660$ \\
SLC2A3 & $0$ & $00438$ & SAMD5 & $-0$ & $00655$ \\
IGFBP1 & $0$ & $00427$ & HSP90B1 & $-0$ & $00641$ \\
A4GALT & $0$ & $00418$ & CFDP1 & $-0$ & $00617$ \\
CEBPB & $0$ & $00404$ & RMND5A & $-0$ & $00614$ \\
PLOD1 & $0$ & $00369$ & CIDEC & $-0$ & $00596$ \\
VSTM2L & $0$ & $00352$ & TPD52L2 & $-0$ & $00579$ \\
XXYLT1 & $0$ & $00341$ & ZNF565 & $-0$ & $00565$ \\
MAP3K8 & $0$ & $00338$ & ACE & $-0$ & $00556$ \\
SNRPB & $0$ & $00276$ & AGRP & $-0$ & $00509$ \\
TOM1L2 & $0$ & $00266$ & PLIN3 & $-0$ & $00506$ \\
NRP2 & $0$ & $00250$ & ARHGEF19 & $-0$ & $00476$ \\
P4HA1 & $0$ & $00225$ & DHRS9 & $-0$ & $00454$ \\
HRASLS2 & $0$ & $00196$ & ATF7IP2 & $-0$ & $00405$ \\
UPP1 & $0$ & $00182$ & NELFE & $-0$ & $00390$ \\
SPP1 & $0$ & $00175$ & RGS3 & $-0$ & $00319$ \\
LAMA5 & $0$ & $00174$ & TNFRSF10D & $-0$ & $00315$ \\
PHACTR3 & $0$ & $00172$ & LOC100506562 & $-0$ & $00290$ \\
ZPLD1 & $0$ & $00165$ & RFX2 & $-0$ & $00264$ \\
CATSPER1 & $0$ & $00163$ & SNORA11D & $-0$ & $00256$ \\
ABHD16A & $0$ & $00143$ & FGB & $-0$ & $00252$ \\
PPP1R3C & $0$ & $00125$ & ICAM2 & $-0$ & $00232$ \\
ADM & $0$ & $00122$ & LGALS9B & $-0$ & $00232$ \\
SOD2 & $0$ & $00120$ & POP5 & $-0$ & $00224$ \\
PPP1R12B & $0$ & $00096$ & NMB & $-0$ & $00205$ \\
NAMPT & $0$ & $00071$ & SERPINB3 & $-0$ & $00201$ \\
KCNQ3 & $0$ & $00040$ & ORC1 & $-0$ & $00199$ \\
MCEMP1 & $0$ & $00025$ & ALOX5AP & $-0$ & $00179$ \\
LYNX1 & $0$ & $00001$ & SLC15A1 & $-0$ & $00139$ \\
\multicolumn{3}{l}{} & OAZ1 & $-0$ & $00134$ \\
\multicolumn{3}{l}{} & TREM1 & $-0$ & $00073$ \\
\multicolumn{3}{l}{} & IKBIP & $-0$ & $00033$ \\
\end{longtable}

\chapter{R code to calculate \texorpdfstring{\acrshort{MSKCC}}{MSKCC} nomogram survival estimates}
\label{app:nomo-mskcc-code}
\begin{lstlisting}[language=R]
fit.mskcc = list(
	inputs = list(
	History.Diagnosis.AgeAt = list(
		margins = data.frame(value = 65, fraction = 1),
		scorefunc = function(x) { x = x; -2/15*pmin(pmax(x, 0), 90) + 12 }),
	Patient.Sex = list(
		margins = data.frame(value = c("M", "F"), fraction = c(0.501, 1-0.501)),
		scorefunc = function(x) { 3*I(x == "M") }),
	Portal.Vein = list(
		margins = data.frame(value = c(TRUE, FALSE), fraction = c(0.144, 1-0.144)),
		scorefunc = function(x) { 10*I(x == TRUE) }),
	Splenectomy = list(
		margins = data.frame(value = c(TRUE, FALSE), fraction = c(0.099, 1-0.099)),
		scorefunc = function(x) { 62*I(x == TRUE) }),
	Treat.MarginPositive = list(
		margins = data.frame(value = c(TRUE, FALSE), fraction = c(0.207, 1-0.207)),
		scorefunc = function(x) { 4*I(x == TRUE) }),
	Path.LocationBody = list(
		margins = data.frame(value = c(FALSE, TRUE), fraction = c(0.894, 1-0.894)),
		scorefunc = function(x) { 51*I(x == TRUE) }),
	Path.Differentiation = list(
		margins = data.frame(value = c("1", "2", "3", "4"), fraction = c(0.142, 0.564, 1-0.142-0.564, 0)),
		scorefunc = function(x) { 14*I(x == "2") + 35*I(x == "3") + 35*I(x == "4") }),		# Undifferentiated (4) not covered by the MSKCC nomogram; here assign the same score as poorly differentiated (3)
	Posterior.Margin = list(
		margins = data.frame(value = c(TRUE, FALSE), fraction = c(0.86, 1-0.86)),
		scorefunc = function(x) { 22*I(x == TRUE) }),
	Path.LN.Involved = list(
		margins = data.frame(value = 2.1, fraction = 1),
		scorefunc = function(x) { 
			x = pmin(40, pmax(x, 0))
			fitfun = splinefun(c(0, 1, 2, 3, 4, 10, 15, 20, 25, 30, 35, 40), c(0, 14.56, 24.64, 30.28, 33.00, 39.05, 43.89, 48.83, 53.77, 58.61, 63.55, 68.49), method = "natural")
			fitfun(x)
		}),
	Path.LN.Negative = list(
		margins = data.frame(value = 16.9, fraction = 1),
		scorefunc = function(x) { (pmin(pmax(x, 0), 90)-90)*-11/90 }),
	Back.pain = list(
		margins = data.frame(value = c(TRUE, FALSE), fraction = c(0.137, 1-0.137)),
		scorefunc = function(x) { 15*I(x == TRUE) }),
	Stage.pT.Simplified = list(
		margins = data.frame(value = c("T1", "T2", "T34"), fraction = c(0.037, 0.119, 1-0.037-0.119)),
		scorefunc = function(x) { 36*I(x == "T1") + 11*I(x == "T34") }),
		# The following matches the original Brennan nomogram, but was not used as there are too few T4
		# tumours in either the NSWPCN *or* the MSKCC cohorts -- how the T4 coefficient was ever estimated,
		# I'll never know.  The T34 coefficient of 11 was arrived at as (0.828*10+(1-0.037-0.119-0.828)*63)/(1-0.037-0.119),
		# being a frequency-weighted average of the T3 and T4 coefficients.
		# margins = data.frame(value = c("T1", "T2", "T3", "T4"), fraction = c(0.037, 0.119, 0.828, 1-0.037-0.119-0.828)),
		# scorefunc = function(x) { 36*I(x == "T1") + 10*I(x == "T3") + 63*I(x == "T4") }),
	Weight.loss = list(
		margins = data.frame(value = c(TRUE, FALSE), fraction = c(0.537, 1-0.537)),
		scorefunc = function(x) { 3*I(x == TRUE) }),
	Path.Size = list(
		margins = data.frame(),
		scorefunc = function(x) {
			x = pmin(16, pmax(x, 0))
			fitfun = splinefun(c(0, 1, 2, 3, 4, 6, 8, 10, 12, 14, 16), c(0, 29.74, 59.48, 86.70, 100, 97.29, 90.03, 82.77, 75.51, 68.25, 61.10), method = "natural")
			fitfun(x)
		}) ),
	outputs = list(
		DSS12mo = function(s) {
			x = pmax(50, pmin(350, s))
			fitfun = splinefun(c(79.0323, 115.02, 165.524, 197.278, 221.774, 242.339, 261.089, 279.839, 299.194, 323.992, 337.298), c(0.94, 0.9, 0.8, 0.7, 0.6, 0.5, 0.4, 0.3, 0.2, 0.1, 0.06))
			y = fitfun(x)
			pmax(0, pmin(1, y))
		},
		DSS24mo = function(s) {
			x = pmax(50, pmin(350, s))
			fitfun = splinefun(c(71.1694, 97.7823, 129.536, 153.73, 174.294, 193.347, 211.794, 231.452, 255.645, 303.125), c(0.86, 0.8, 0.7, 0.6, 0.5, 0.4, 0.3, 0.2, 0.1, 0.01))
			y = fitfun(x)
			pmax(0, pmin(1, y))
		},
		DSS36mo = function(s) {
			x = pmax(50, pmin(350, s))
			fitfun = splinefun(c(69.3548, 101.109, 125.302, 145.867, 164.919, 183.367, 202.722, 226.915, 274.093), c(0.8, 0.7, 0.6, 0.5, 0.4, 0.3, 0.2, 0.1, 0.01))
			y = fitfun(x)
			pmax(0, pmin(1, y))
		}) 
	)

applyNomogram = function(nomogram, data)
{
	scores = rowSums(sapply(names(nomogram$inputs), function(input) {
		if (input %in% colnames(data)) {
			return(nomogram$inputs[[input]]$scorefunc(data[,input]))
		}
		warning(sprintf("Marginalizing missing variable: %s", input))
		margin_score = sum(nomogram$inputs[[input]]$scorefunc(nomogram$inputs[[input]]$margins$value) * nomogram$inputs[[input]]$margins$fraction)
		return(rep(margin_score, nrow(data)))
	}))

	outputs = sapply(nomogram$outputs, function(f) f(scores))
	cbind(Score = scores, outputs)
}
\end{lstlisting}
