\appendix
\appendixpage*
\addappheadtotoc

\chapter{R code to calculate \acrshort{MSKCC} nomogram survival estimates}
\label{app:nomo-mskcc-code}
\begin{lstlisting}[language=R]
fit.mskcc = list(
	inputs = list(
	History.Diagnosis.AgeAt = list(
		margins = data.frame(value = 65, fraction = 1),
		scorefunc = function(x) { x = x; -2/15*pmin(pmax(x, 0), 90) + 12 }),
	Patient.Sex = list(
		margins = data.frame(value = c("M", "F"), fraction = c(0.501, 1-0.501)),
		scorefunc = function(x) { 3*I(x == "M") }),
	Portal.Vein = list(
		margins = data.frame(value = c(TRUE, FALSE), fraction = c(0.144, 1-0.144)),
		scorefunc = function(x) { 10*I(x == TRUE) }),
	Splenectomy = list(
		margins = data.frame(value = c(TRUE, FALSE), fraction = c(0.099, 1-0.099)),
		scorefunc = function(x) { 62*I(x == TRUE) }),
	Treat.MarginPositive = list(
		margins = data.frame(value = c(TRUE, FALSE), fraction = c(0.207, 1-0.207)),
		scorefunc = function(x) { 4*I(x == TRUE) }),
	Path.LocationBody = list(
		margins = data.frame(value = c(FALSE, TRUE), fraction = c(0.894, 1-0.894)),
		scorefunc = function(x) { 51*I(x == TRUE) }),
	Path.Differentiation = list(
		margins = data.frame(value = c("1", "2", "3", "4"), fraction = c(0.142, 0.564, 1-0.142-0.564, 0)),
		scorefunc = function(x) { 14*I(x == "2") + 35*I(x == "3") + 35*I(x == "4") }),		# Undifferentiated (4) not covered by the MSKCC nomogram; here assign the same score as poorly differentiated (3)
	Posterior.Margin = list(
		margins = data.frame(value = c(TRUE, FALSE), fraction = c(0.86, 1-0.86)),
		scorefunc = function(x) { 22*I(x == TRUE) }),
	Path.LN.Involved = list(
		margins = data.frame(value = 2.1, fraction = 1),
		scorefunc = function(x) { 
			x = pmin(40, pmax(x, 0))
			fitfun = splinefun(c(0, 1, 2, 3, 4, 10, 15, 20, 25, 30, 35, 40), c(0, 14.56, 24.64, 30.28, 33.00, 39.05, 43.89, 48.83, 53.77, 58.61, 63.55, 68.49), method = "natural")
			fitfun(x)
		}),
	Path.LN.Negative = list(
		margins = data.frame(value = 16.9, fraction = 1),
		scorefunc = function(x) { (pmin(pmax(x, 0), 90)-90)*-11/90 }),
	Back.pain = list(
		margins = data.frame(value = c(TRUE, FALSE), fraction = c(0.137, 1-0.137)),
		scorefunc = function(x) { 15*I(x == TRUE) }),
	Stage.pT.Simplified = list(
		margins = data.frame(value = c("T1", "T2", "T34"), fraction = c(0.037, 0.119, 1-0.037-0.119)),
		scorefunc = function(x) { 36*I(x == "T1") + 11*I(x == "T34") }),
		# The following matches the original Brennan nomogram, but was not used as there are too few T4
		# tumours in either the NSWPCN *or* the MSKCC cohorts -- how the T4 coefficient was ever estimated,
		# I'll never know.  The T34 coefficient of 11 was arrived at as (0.828*10+(1-0.037-0.119-0.828)*63)/(1-0.037-0.119),
		# being a frequency-weighted average of the T3 and T4 coefficients.
		# margins = data.frame(value = c("T1", "T2", "T3", "T4"), fraction = c(0.037, 0.119, 0.828, 1-0.037-0.119-0.828)),
		# scorefunc = function(x) { 36*I(x == "T1") + 10*I(x == "T3") + 63*I(x == "T4") }),
	Weight.loss = list(
		margins = data.frame(value = c(TRUE, FALSE), fraction = c(0.537, 1-0.537)),
		scorefunc = function(x) { 3*I(x == TRUE) }),
	Path.Size = list(
		margins = data.frame(),
		scorefunc = function(x) {
			x = pmin(16, pmax(x, 0))
			fitfun = splinefun(c(0, 1, 2, 3, 4, 6, 8, 10, 12, 14, 16), c(0, 29.74, 59.48, 86.70, 100, 97.29, 90.03, 82.77, 75.51, 68.25, 61.10), method = "natural")
			fitfun(x)
		}) ),
	outputs = list(
		DSS12mo = function(s) {
			x = pmax(50, pmin(350, s))
			fitfun = splinefun(c(79.0323, 115.02, 165.524, 197.278, 221.774, 242.339, 261.089, 279.839, 299.194, 323.992, 337.298), c(0.94, 0.9, 0.8, 0.7, 0.6, 0.5, 0.4, 0.3, 0.2, 0.1, 0.06))
			y = fitfun(x)
			pmax(0, pmin(1, y))
		},
		DSS24mo = function(s) {
			x = pmax(50, pmin(350, s))
			fitfun = splinefun(c(71.1694, 97.7823, 129.536, 153.73, 174.294, 193.347, 211.794, 231.452, 255.645, 303.125), c(0.86, 0.8, 0.7, 0.6, 0.5, 0.4, 0.3, 0.2, 0.1, 0.01))
			y = fitfun(x)
			pmax(0, pmin(1, y))
		},
		DSS36mo = function(s) {
			x = pmax(50, pmin(350, s))
			fitfun = splinefun(c(69.3548, 101.109, 125.302, 145.867, 164.919, 183.367, 202.722, 226.915, 274.093), c(0.8, 0.7, 0.6, 0.5, 0.4, 0.3, 0.2, 0.1, 0.01))
			y = fitfun(x)
			pmax(0, pmin(1, y))
		}) 
	)

applyNomogram = function(nomogram, data)
{
	scores = rowSums(sapply(names(nomogram$inputs), function(input) {
		if (input %in% colnames(data)) {
			return(nomogram$inputs[[input]]$scorefunc(data[,input]))
		}
		warning(sprintf("Marginalizing missing variable: %s", input))
		margin_score = sum(nomogram$inputs[[input]]$scorefunc(nomogram$inputs[[input]]$margins$value) * nomogram$inputs[[input]]$margins$fraction)
		return(rep(margin_score, nrow(data)))
	}))

	outputs = sapply(nomogram$outputs, function(f) f(scores))
	cbind(Score = scores, outputs)
}
\end{lstlisting}

\chapter{Basis matrix $W$ for the six survival-associated metagenes}
\label{app:sigs-w-matrix}
\input{analysis/biosurv/reports/18_SIS_diag_dsd_final/basis_table.tex}

\chapter{\acrshort{MSigDB} signatures correlated with axis A1}
\label{app:sigs-msigdb-corrs-axis1}
\begin{table}[!htbp]
\centering
\caption[\acrshort{MSigDB} signatures correlated with axis A1]{\acrshort{MSigDB} signatures substantially correlated with activity of the prognostic axis A1.}
\begin{tabular}{@{}ll@{}}
\toprule
MSigDB set                                                                                                                                                                                                                              & A1 correlation \\ \midrule
c5.M\_PHASE/c5.MITOSIS/c5.M\_PHASE\_OF\_MITOTIC\_CELL\_CYCLE                                                                                                                                                                            & 0.689          \\
c5.REGULATION\_OF\_MITOSIS                                                                                                                                                                                                              & 0.682          \\
c4.GNF2\_RFC3/c4.GNF2\_RFC4/c4.GNF2\_SMC2L1/c4.GNF2\_CKS1B/c4.GNF2\_CKS2/c4.GNF2\_TTK                                                                                                                                                   & 0.664          \\
c5.CELL\_CYCLE\_PROCESS/c5.MITOTIC\_CELL\_CYCLE/c5.CELL\_CYCLE\_PHASE                                                                                                                                                                   & 0.653          \\
c5.SPINDLE                                                                                                                                                                                                                              & 0.644          \\
c4.MORF\_BUB1B                                                                                                                                                                                                                          & 0.631          \\
c6.CSR\_LATE\_UP.V1\_SIGNED                                                                                                                                                                                                             & 0.630          \\
c5.SPINDLE\_POLE                                                                                                                                                                                                                        & 0.628          \\
c2.PID\_PLK1\_PATHWAY                                                                                                                                                                                                                   & 0.626          \\
c5.ORGANELLE\_PART/c5.INTRACELLULAR\_ORGANELLE\_PART                                                                                                                                                                                    & 0.624          \\
c2.REACTOME\_CELL\_CYCLE/c2.REACTOME\_CELL\_CYCLE\_MITOTIC                                                                                                                                                                              & 0.622          \\
c2.REACTOME\_CYCLIN\_A\_B1\_ASSOCIATED\_EVENTS\_DURING\_G2\_M\_TRANSITION                                                                                                                                                               & 0.604          \\
c2.REACTOME\_MITOTIC\_PROMETAPHASE                                                                                                                                                                                                      & 0.596          \\
c2.KEGG\_CELL\_CYCLE                                                                                                                                                                                                                    & 0.588          \\
c5.CHROMOSOME\_SEGREGATION                                                                                                                                                                                                              & 0.588          \\
c4.MORF\_FEN1                                                                                                                                                                                                                           & 0.586          \\
c2.REACTOME\_G1\_S\_SPECIFIC\_TRANSCRIPTION                                                                                                                                                                                             & 0.585          \\
c2.REACTOME\_ACTIVATION\_OF\_THE\_PRE\_REPLICATIVE\_COMPLEX/c2.REACTOME\_ACTIVATION\_OF\_ATR\_IN\_RESPONSE\_TO\_REPLICATION\_STRESS/c2.REACTOME\_G2\_M\_CHECKPOINTS                                                                     & 0.583          \\
c2.REACTOME\_E2F\_ENABLED\_INHIBITION\_OF\_PRE\_REPLICATION\_COMPLEX\_FORMATION                                                                                                                                                         & 0.581          \\
c2.REACTOME\_E2F\_MEDIATED\_REGULATION\_OF\_DNA\_REPLICATION                                                                                                                                                                            & 0.577          \\
c5.CELL\_CYCLE\_GO\_0007049                                                                                                                                                                                                             & 0.576          \\
c2.REACTOME\_KINESINS                                                                                                                                                                                                                   & 0.575          \\
c3.V\$ELK1\_02                                                                                                                                                                                                                          & 0.574          \\
c5.SPINDLE\_MICROTUBULE                                                                                                                                                                                                                 & 0.573          \\
c5.MITOTIC\_CELL\_CYCLE\_CHECKPOINT                                                                                                                                                                                                     & 0.569          \\
c2.REACTOME\_CELL\_CYCLE\_CHECKPOINTS/c2.REACTOME\_G1\_S\_TRANSITION/c2.REACTOME\_SYNTHESIS\_OF\_DNA/c2.REACTOME\_MITOTIC\_G1\_G1\_S\_PHASES/c2.REACTOME\_MITOTIC\_M\_M\_G1\_PHASES/c2.REACTOME\_DNA\_REPLICATION/c2.REACTOME\_S\_PHASE & 0.566          \\
c4.MORF\_ESPL1                                                                                                                                                                                                                          & 0.566          \\
c4.MORF\_BUB1                                                                                                                                                                                                                           & 0.565          \\
c4.MORF\_BUB3/c4.MORF\_RAD23A                                                                                                                                                                                                           & 0.563          \\
c5.CONDENSED\_CHROMOSOME                                                                                                                                                                                                                & 0.562          \\
c4.MORF\_RFC4/c4.MORF\_RRM1                                                                                                                                                                                                             & 0.561          \\
c2.BIOCARTA\_G2\_PATHWAY                                                                                                                                                                                                                & 0.559          \\
c3.SCGGAAGY\_V\$ELK1\_02                                                                                                                                                                                                                & 0.558          \\
c2.PID\_AURORA\_A\_PATHWAY                                                                                                                                                                                                              & 0.556          \\
c5.MITOTIC\_SISTER\_CHROMATID\_SEGREGATION/c5.SISTER\_CHROMATID\_SEGREGATION                                                                                                                                                            & 0.555          \\
c4.MORF\_UNG                                                                                                                                                                                                                            & 0.554          \\
c2.PID\_FOXM1PATHWAY                                                                                                                                                                                                                    & 0.551          \\
c4.MORF\_GSPT1                                                                                                                                                                                                                          & 0.550          \\
c2.REACTOME\_METABOLISM\_OF\_NUCLEOTIDES                                                                                                                                                                                                & 0.550          \\
c2.PID\_ATR\_PATHWAY                                                                                                                                                                                                                    & 0.547          \\
c2.BIOCARTA\_MCM\_PATHWAY                                                                                                                                                                                                               & 0.546          \\
c4.MORF\_CCNF                                                                                                                                                                                                                           & 0.544          \\
c5.CELL\_CYCLE\_CHECKPOINT\_GO\_0000075                                                                                                                                                                                                 & 0.543          \\
c5.MITOTIC\_SPINDLE\_ORGANIZATION\_AND\_BIOGENESIS/c5.SPINDLE\_ORGANIZATION\_AND\_BIOGENESIS                                                                                                                                            & 0.542          \\
c4.MORF\_EI24                                                                                                                                                                                                                           & 0.538          \\
c5.DOUBLE\_STRAND\_BREAK\_REPAIR                                                                                                                                                                                                        & 0.537          \\
c4.GNF2\_PA2G4/c4.GNF2\_RAN                                                                                                                                                                                                             & 0.531          \\
c2.REACTOME\_G2\_M\_DNA\_DAMAGE\_CHECKPOINT                                                                                                                                                                                             & 0.531          \\
c2.KEGG\_PYRIMIDINE\_METABOLISM                                                                                                                                                                                                         & 0.531          \\
c4.MORF\_GMPS                                                                                                                                                                                                                           & 0.528          \\
c4.MORF\_PRKDC                                                                                                                                                                                                                          & 0.528          \\
c2.PID\_BARD1PATHWAY                                                                                                                                                                                                                    & 0.528          \\
c4.GNF2\_MCM5                                                                                                                                                                                                                           & 0.525          \\
c4.MORF\_DNMT1                                                                                                                                                                                                                          & 0.524          \\
c2.REACTOME\_POL\_SWITCHING                                                                                                                                                                                                             & 0.523          \\
c4.GNF2\_MSH2                                                                                                                                                                                                                           & 0.521          \\
c4.MORF\_CSNK2B                                                                                                                                                                                                                         & 0.520          \\
c2.PID\_AURORA\_B\_PATHWAY                                                                                                                                                                                                              & 0.520          \\
c2.REACTOME\_DESTABILIZATION\_OF\_MRNA\_BY\_KSRP                                                                                                                                                                                        & 0.517          \\
c5.DNA\_METABOLIC\_PROCESS                                                                                                                                                                                                              & 0.517          \\
c4.MORF\_PTPN11                                                                                                                                                                                                                         & 0.516          \\
c5.REGULATION\_OF\_MITOTIC\_CELL\_CYCLE                                                                                                                                                                                                 & 0.516          \\
c5.RESPONSE\_TO\_ENDOGENOUS\_STIMULUS/c5.RESPONSE\_TO\_DNA\_DAMAGE\_STIMULUS                                                                                                                                                            & 0.515          \\
c5.CHROMOSOMEPERICENTRIC\_REGION/c5.KINETOCHORE                                                                                                                                                                                         & 0.514          \\
c6.MTOR\_UP.V1\_SIGNED                                                                                                                                                                                                                  & 0.512          \\
c2.REACTOME\_APOPTOSIS                                                                                                                                                                                                                  & 0.510          \\
c4.MORF\_PPP1CC                                                                                                                                                                                                                         & 0.509          \\
c5.PORE\_COMPLEX/c5.NUCLEAR\_PORE                                                                                                                                                                                                       & 0.508          \\
c5.DNA\_REPAIR                                                                                                                                                                                                                          & 0.506          \\
c2.REACTOME\_CHROMOSOME\_MAINTENANCE/c2.REACTOME\_TELOMERE\_MAINTENANCE                                                                                                                                                                 & 0.506          \\
c5.MACROMOLECULAR\_COMPLEX/c5.PROTEIN\_COMPLEX                                                                                                                                                                                          & 0.506          \\
c4.MORF\_XRCC5/c4.MORF\_GNB1                                                                                                                                                                                                            & 0.504          \\
c5.INTERPHASE/c5.INTERPHASE\_OF\_MITOTIC\_CELL\_CYCLE                                                                                                                                                                                   & 0.503          \\
c5.NON\_MEMBRANE\_BOUND\_ORGANELLE/c5.INTRACELLULAR\_NON\_MEMBRANE\_BOUND\_ORGANELLE                                                                                                                                                    & 0.503          \\
c6.GCNP\_SHH\_UP\_EARLY.V1\_SIGNED                                                                                                                                                                                                      & 0.503          \\
c2.BIOCARTA\_RANMS\_PATHWAY                                                                                                                                                                                                             & 0.502          \\
c2.KEGG\_DNA\_REPLICATION/c2.REACTOME\_DNA\_STRAND\_ELONGATION                                                                                                                                                                          & 0.502          \\
c4.MORF\_SOD1                                                                                                                                                                                                                           & 0.502          \\
c5.NUCLEAR\_MEMBRANE/c5.NUCLEAR\_MEMBRANE\_PART                                                                                                                                                                                         & 0.501          \\
c4.MORF\_HDAC1                                                                                                                                                                                                                          & 0.501          \\
c2.REACTOME\_HIV\_LIFE\_CYCLE/c2.REACTOME\_LATE\_PHASE\_OF\_HIV\_LIFE\_CYCLE                                                                                                                                                            & 0.500          \\
c5.CHROMOSOMAL\_PART/c5.CHROMOSOME                                                                                                                                                                                                      & 0.500          \\
c5.PHOSPHORIC\_DIESTER\_HYDROLASE\_ACTIVITY                                                                                                                                                                                             & -0.502         \\
c3.CTGCAGY\_UNKNOWN                                                                                                                                                                                                                     & -0.505         \\
c3.V\$OCT1\_01                                                                                                                                                                                                                          & -0.509         \\
c3.V\$GATA\_Q6                                                                                                                                                                                                                          & -0.515         \\
c5.CELL\_SURFACE\_RECEPTOR\_LINKED\_SIGNAL\_TRANSDUCTION\_GO\_0007166                                                                                                                                                                   & -0.518         \\
c4.GNF2\_MAPT                                                                                                                                                                                                                           & -0.526         \\
c3.V\$OCT1\_04                                                                                                                                                                                                                          & -0.531         \\
c2.REACTOME\_G\_ALPHA\_S\_SIGNALLING\_EVENTS                                                                                                                                                                                            & -0.539         \\
c3.V\$OCT\_C                                                                                                                                                                                                                            & -0.544         \\ \bottomrule
\end{tabular}
\end{table}

\chapter{\acrshort{MSigDB} signatures correlated with axis A2}
\label{app:sigs-msigdb-corrs-axis2}
\begin{table}[!htbp]
\centering
\caption[\acrshort{MSigDB} signatures correlated with axis A2]{\acrshort{MSigDB} signatures substantially correlated with activity of the prognostic axis A2.}
\begin{tabular}{@{}ll@{}}
\toprule
GeneSet                                                                           & Correlation \\ \midrule
c2.PID\_INTEGRIN1\_PATHWAY                                                        & 0.654       \\
c2.PID\_INTEGRIN3\_PATHWAY                                                        & 0.637       \\
c2.PID\_UPA\_UPAR\_PATHWAY                                                        & 0.597       \\
c4.GNF2\_PTX3                                                                     & 0.593       \\
c2.KEGG\_ECM\_RECEPTOR\_INTERACTION                                               & 0.582       \\
c2.PID\_INTEGRIN5\_PATHWAY                                                        & 0.577       \\
c4.GNF2\_MMP1                                                                     & 0.575       \\
c2.REACTOME\_EXTRACELLULAR\_MATRIX\_ORGANIZATION/c2.REACTOME\_COLLAGEN\_FORMATION & 0.572       \\
c5.AXON\_GUIDANCE                                                                 & 0.571       \\
c2.KEGG\_FOCAL\_ADHESION                                                          & 0.567       \\
c2.PID\_SYNDECAN\_1\_PATHWAY                                                      & 0.552       \\
c2.REACTOME\_CELL\_EXTRACELLULAR\_MATRIX\_INTERACTIONS                            & 0.538       \\
c2.PID\_INTEGRIN\_CS\_PATHWAY                                                     & 0.536       \\
c5.TISSUE\_DEVELOPMENT                                                            & 0.536       \\
c5.COLLAGEN                                                                       & 0.531       \\
c6.CORDENONSI\_YAP\_CONSERVED\_SIGNATURE                                          & 0.526       \\
c6.LEF1\_UP.V1\_SIGNED                                                            & 0.519       \\
c2.REACTOME\_INTEGRIN\_CELL\_SURFACE\_INTERACTIONS                                & 0.518       \\
c5.AXONOGENESIS/c5.CELLULAR\_MORPHOGENESIS\_DURING\_DIFFERENTIATION               & 0.515       \\
c6.STK33\_NOMO\_SIGNED                                                            & 0.507       \\
c7.GSE17721\_CTRL\_VS\_CPG\_12H\_BMDM\_SIGNED                                     & -0.508      \\
c7.GSE1460\_INTRATHYMIC\_T\_PROGENITOR\_VS\_THYMIC\_STROMAL\_CELL\_SIGNED         & -0.508      \\ \bottomrule
\end{tabular}
\end{table}


\chapter{Approximate calculation of \acrshort{PARSE} scores}
\label{app:sigs-parse-approx}
Exact calculation of \gls{PARSE} score requires the solution of a number of \gls{NNLS} problems, which complicates application.  The \gls{NNLS} solutions can be approximated with conventional least squares solutions, ultimately transforming the calculation of an approximate \gls{PARSE} score into a simple weighted sum of gene expression measurements.

Recall that \gls{NMF} finds factorizations of the form $A = W H$, with all elements of $A$, $W$, and $H$, being non-negative.  In the reverse problem of \gls{PARSE} calculation, $A$ and $\widehat{W}$ are supplied, and $H$ is to be estimated.  I propose an approximation that removes the requirement that $H$ be non-negative, $H \approx \widehat{W}^+ A$, where $\widehat{W}^+$ is the Moore-Penrose pseudoinverse of $\widehat{W}$.  By combining this approximation with the linear combination of metagene coefficients that forms the \gls{PARSE} score, we can approximate \gls{PARSE} as a simple weighted sum of gene expression measurements:
\begin{align}
  P &= L H \\
    &\approx L \widehat{W}^+ A \\
    &= k A
\end{align}
where $P$ is the vector of \gls{PARSE} score values, $L$ is the metagene loadings for the \gls{PARSE} score, $L = \irow{1.354&-1.548&0&0&-1.354&1.548}$, and $k$ is a row vector of gene loadings for calculation of an approximate \gls{PARSE} score.  Approximation of $P$ by $k A$ appears excellent; when tested on \gls{APGI} gene expression measurements, the approximation closely matched the more laborious exact \gls{NNLS} solution (\fref{fig:app-parse-approx-matching}).

\begin{figure}
\centering
\includegraphics[width=.7\linewidth]{analysis/biosurv/reports/18_SIS_diag_dsd_final/figure/approx-calc-1}
\caption[Performance of the \acrshort{PARSE} score approximation]{The linear \acrshort{PARSE} score approximation $P \approx k A$ closely matches the exact version calculated using \gls{NNLS}, when evaluated on \gls{APGI} \gls{GEX} data.}\label{fig:app-parse-approx-matching}
\end{figure}

To use the approximation in practice, perform the following steps:
\begin{enumerate}
  \item Prepare a gene $\times$ sample matrix of linear expression estimates $A$, in which values for each row (gene) have been scaled to encompass the range $0$ to $1$.
  \item Subset $A$ to only the genes present in the $k$ table (below), and arrange rows of $A$ so that they exactly match the order of rows of $k$.  If genes present in $k$ are missing from $A$, insert all-zero rows for these genes into $A$.
  \item Calculate approximate \gls{PARSE} scores $P$ as $P = k A$.  This is equivalent to, for each column (sample) of $A$, multiplying each entry of the column of $A$ with the corresponding entry of $k$, and summing the results.
\end{enumerate}

The loading vector for the calculation of approximate \gls{PARSE} score, $k^T$, follows.
\input{analysis/biosurv/reports/18_SIS_diag_dsd_final/approx_table.tex}
