\documentclass[thesis.tex]{subfiles}
\begin{document}
\chapter[Conclusion]{Conclusion}
\label{chap:conc}

\section*{}

\Acrlong{PDAC} is a devastating cancer that is remarkably refractory to all current therapies, and there is a real need for better understanding and management of this disease.  Recent genomic efforts have begun to exhaustively characterise \gls{PDAC} at the molecular level, but the analysis of the data generated by these projects has only just begun.  This dissertation reported on one aspect of this analysis, by exploring the link between gene expression and \gls{PDAC} outcome, in ways that were previously not possible.  It yielded a better understanding of the basic biological processes that reflect, and possibly control, survival in \gls{PDAC}, and developed tools to assist the management of the disease, with the goal of ultimately improving patient quality of life.  Beyond the application to \gls{PDAC} presented here, the analysis frameworks used in this dissertation might find use in exploring the links between gene expression and prognosis in other diseases.

\subsection{\texorpdfstring{\Cref{chap:signatures}}{Chapter 2}: Signatures of survival}
\Cref{chap:signatures} addressed a simple question: what fundamental biological processes are linked to differences in the survival of patients with resected \gls{PDAC}?  By identifying genes with prognostic expression, and then factorising these genes into coordinately-expressed metagenes, I identified two main orthogonal axes of transcription in \gls{PDAC} samples, that independently were associated with patient survival time.  These two axes were closely correlated with previously-reported signatures of cell proliferation and the \gls{EMT}, strongly implicating these two processes in the differential survival of patients with \gls{PDAC}.  This result makes sense in light of the known biology of resected \gls{PDAC}, where it is believed that the majority of resected patients have disseminated undetected metastases at the time of resection, which lead to early relapse and death~\cite{Barugola2007}.  In this light, the importance of \gls{EMT} (causing metastasis), and proliferation (general malignancy), appear reasonable, but the absence of other aspects of \gls{PDAC} biology from the axes was surprising.

The many aspects of cancer have been conceptually grouped into ten so-called hallmarks~\cite{Hanahan2011}, with many contributing to a given malignancy; in this light, it is surprising that only two hallmarks were associated with outcome in \gls{PDAC}.  In particular, the absence of any strong stromal signatures, or any significant association between tumour stromal content and expression, was surprising given the importance of the stroma to \gls{PDAC} pathology~\cite{Luo2012}.  It may be that the stroma is in fact involved in \gls{PDAC} prognosis, but not in \emph{differential} prognosis; the work of \Cref{chap:signatures} was powered only to detect the latter.

Both transcriptional axes identified in \Cref{chap:signatures} were prognostic in a number of other solid tumours, often with distinct patterns.  For example, both axes were approximately additive on hazard in \gls{PDAC}, but interacted in kidney renal clear cell carcinoma, in which high expression on \emph{both} the proliferation and \gls{EMT} axes was required for poor outcome.  This hints at both commonality and differences in the importance of these axes to outcome in cancer in general, and provides further evidence that these axes are truly associated with outcome.  It also suggests that the methods developed in \Cref{chap:signatures} could have broad application.

\Cref{chap:signatures} developed a template for the principled analysis of transcription patterns linked to differential outcome, in any sample type.  Many of the methods used there are new (for example, \gls{CPSS}), or difficult to apply correctly (\gls{NMF}), but the success of the chapter's work suggests that derivative analyses following the template developed in \Cref{chap:signatures} in other diseases will be equally fruitful.  Of particular interest would be the application of the methods of \Cref{chap:signatures} to the pan-cancer datasets that are now reaching maturity; this endeavour could provide a unique window on those metagenes that are survival-associated across many cancers, and perhaps lead to better understanding of the connections between different malignancies.

Better understanding of the processes that drive survival in cancer could point the way to more effective therapy.  The field of \gls{PDAC} in particular is in desperate need of more effective ways to treat the disease, with all the tools of molecular medicine so far having made only a modest influence on overall survival with the cancer.  If the survival-associated axes in \gls{PDAC} are in fact causative of outcome, then methods to modulate these axes may permit the transformation of a poor-prognosis disease into a good-prognosis one.  For a cancer as aggressive as \gls{PDAC} this manipulation is unlikely to affect a cure, but even small gains are positive news for this dire malignancy.

\subsection{\texorpdfstring{\Cref{chap:nomogram}}{Chapter 3}: Predicting outcome}
\Cref{chap:nomogram} addressed a pressing problem in the management of \gls{PDAC}: resection of the primary tumour is a major operation that carries risk and morbidity, and it is offered to all patients with a resectable tumour, yet relatively few patients derive much benefit from the procedure.  Most of the patients who undergo pancreas resection for \gls{PDAC} will have undetected metastatic disease, and after taking into account the decreased quality of life following their operation, may actually have been better served by undergoing no resection at all.  The problem is fundamentally one of staging: current staging methods for \gls{PDAC} cannot reliably identify pre-resection patients with micro metastases from those without, and so all eligible patients are offered the resection procedure.

As a first step in resolving this serious problem with \gls{PDAC} management, \Cref{chap:nomogram} described the construction of the \gls{PCOP}, a tool to \emph{preoperatively} predict the expected survival of a \gls{PDAC} patient if they were to undergo resection.  The \gls{PCOP} used preoperative measurements of two biomarkers of metastasis, to be collected during \gls{EUS}-\gls{FNA} performed prior to resection.  Although the prognostic accuracy of the \gls{PCOP} was modest, it was competitive with a gold-standard \gls{PDAC} prognostic nomogram that requires \emph{postoperative} data~\cite{Brennan2004}, and so may still have clinical utility.  In particular, the \gls{PCOP} may be useful as a clinical `tie-breaker,' assisting the patient and surgeon to decide on a course to take with a resection that is expected to be particularly challenging.  In the case of a poor prognosis from \gls{PCOP}, the decision may be made to not resect, as little benefit is expected to be gained.  On the other hand, should the \gls{PCOP} predict good survival following resection, the surgeon and patient may consider a more risky and challenging resection to be worthwhile.  The \gls{PCOP} developed in \Cref{chap:nomogram} is a first version, and is expected to improve over time; in particular, the identification of better biomarkers of prognosis may drastically improve the \gls{PCOP}'s accuracy.
  
\subsection{\texorpdfstring{\Cref{chap:messina}}{Chapter 4}: Better biomarkers}
Many diagnostic and prognostic classifiers don't successfully validate.  This is a serious concern when translating a discovery made in one context, such as a research laboratory, to another, such as a molecular pathology test, with much time and resources wasted on a failed translation project.  To assist with this problem, \Cref{chap:messina} developed and characterised Messina2, a technique for identifying optimally robust classifiers, that can be used for both diagnostic and prognostic separation of patients.  Messina2 significantly outperformed competing methods in both diagnosis and prognosis, and was applied to a resected \gls{PDAC} cohort to identify three mRNAs that strongly separated good- and poor- prognosis patients.  Testing of the expression of protein linked to these mRNAs may reveal improved biomarkers of \gls{PDAC} outcome to those used in \Cref{chap:nomogram}.

\subsection{Final words}
The prognosis of patients with \gls{PDAC} has remained consistently dire for the last 30 years.  This situation may soon change: new genomics efforts are collecting unprecedented numbers of \gls{PDAC} patients, and characterising their disease in great detail, both at the molecular and clinical level.  The challenge will soon be to analyse the huge amount of data from these projects, and the many like them for other diseases, to gain insights on disease that translate to actual improved patient outcomes.  This dissertation describes some early efforts in this direction, by analysing the link between gene expression and outcome, at a number of levels.  Some results are presented that may add usefully to the knowledge of \gls{PDAC}, but I believe that the greater value of this work is the methods that are developed within, that I hope will find some use in furthering future knowledge of disease and health.

\end{document}
