\newacronym{HMM}{HMM}{hidden Markov model}
\newacronym{CNV}{CNV}{copy number variation}
\newacronym{LOH}{LOH}{loss of heterozygosity}
\newacronym{FDR}{FDR}{false-discovery rate}
\newacronym{BIC}{BIC}{Bayesian information criterion}
\newacronym{AIC}{AIC}{Akaike information criterion}
\newacronym{CPV}{CPV}{clinico-pathological variable}
\newacronym{SNV}{SNV}{single nucleotide variant}
\newacronym{LABQSR}{LA-BQSR}{local alignment and base quality score recalibration}
\newacronym{GATK}{GATK}{Genome analysis toolkit}
\newacronym{BAM}{BAM}{binary sequence alignment / map file}
\newacronym{indel}{indel}{insertion / deletion event}
\newacronym{BMC}{BMC}{Bayesian model comparison}
\newacronym{NGS}{NGS}{next-generation sequencing}
\newacronym{WGS}{WGS}{whole genome sequencing}
\newacronym{SNR}{SNR}{signal-to-noise ratio}
\newacronym{KDE}{KDE}{kernel density estimate}
\newacronym{MDS}{MDS}{multidimensional scaling}
\newacronym{IDAT}{IDAT}{Illumina data}
\newacronym{SIS}{SIS}{sure independence screening}
\newacronym{CPSS}{CPSS}{complementary pair subset selection}
\newacronym{FAST}{FAST}{feature aberration at survival times}
\newacronym{NMF}{NMF}{non-negative matrix factorisation}
\newacronym{GEX}{GEX}{gene expression}
\newacronym{VST}{VST}{variance stabilising transform}
\newacronym{GSVA}{GSVA}{gene set variation analysis}
\newacronym{SNMFL}{SNMF/L}{sparse non-negative matrix factorisation, long variant}
\newacronym{SNMFW}{SNMF/W}{sparse non-negative matrix factorisation, wide variant}
\newacronym{MSIGDB}{MSigDB}{molecular signatures database}
\newacronym{LASSO}{LASSO}{least absolute shrinkage and selection operator}
\newacronym{EMT}{EMT}{epithelial to mesenchymal transition}
\newacronym{NNLS}{NNLS}{non-negative least squares}
\newacronym{APGI}{APGI}{Australian Pancreatic Cancer Genome Initiative}
\newacronym{PDAC}{PDAC}{pancreatic ductal adenocarcinoma}
\newacronym{NCBI}{NCBI}{National Center for Biotechnology Information}
\newacronym{GEO}{GEO}{Gene Expression Omnibus}
\newacronym{BH}{BH}{Benjamini-Hochberg}
\newacronym{ECM}{ECM}{extracellular matrix}
\newacronym{CV}{CV}{cross-validation}
\newacronym{TCGA}{TCGA}{The Cancer Genome Atlas}
\newacronym{FWER}{FWER}{familywise error rate}
\newacronym{PARSE}{PARSE}{prognostic axis risk stratification estimate}
\newacronym{DSS}{DSS}{disease-specific survival}
\newacronym{DSD}{DSD}{disease-specific death}
\newacronym{PCA}{PCA}{principal component analysis}
\newacronym{ICA}{ICA}{independent component analysis}
\newacronym{ICGC}{ICGC}{International Cancer Genome Consortium}
\newacronym{CA-19-9}{CA-19-9}{carbohydrate antigen 19-9}
\newacronym{MSKCC}{MSKCC}{Memorial Sloan-Kettering Cancer Center}
\newacronym{EUS}{EUS}{endoscopic ultrasound}
\newacronym{FNA}{FNA}{fine needle aspirate}
\newacronym{IHCry}{IHC}{immunohistochemistry}
\newacronym{IHCal}{IHC}{immunohistochemical}
\newacronym{PPV}{PPV}{positive predictive value}
\newacronym{CT}{CT}{computed tomography}
\newacronym{PH}{PH}{proportional hazard}
\newacronym{CPH}{CPH}{Cox proportional hazard}
\newacronym{AFT}{AFT}{accelerated failure time}
\newacronym{NSWPCN}{NSWPCN}{New South Wales Pancreatic Cancer Network}
\newacronym{LOESS}{LOESS}{local regression}
\newacronym{GG}{GG}{generalised gamma}
\newacronym{KM}{KM}{Kaplan-Meier}
\newacronym{IBS}{IBS}{integrated Brier score}
\newacronym{RSF}{RSF}{random survival forest}
\newacronym{PCOP}{PCOP}{the Pancreas Cancer Outcome Predictor}
\newacronym{ROC}{ROC}{receiver operating characteristic}
\newacronym{AUC}{AUC}{area under the curve}
\newacronym{BAF}{BAF}{B allele frequency}
\newacronym{SNP}{SNP}{single nucleotide polymorphism}
\newacronym{TDROC}{TD-ROC}{time-dependent receiver operating characteristic}
\newacronym{PI}{PI}{prognostic index}
\newacronym{AWS}{AWS}{Amazon Web Services}
\newacronym{SVM}{SVM}{support vector machine}
\newacronym{ECDF}{eCDF}{empirical cumulative distribution function}
\newacronym{CDF}{CDF}{cumulative distribution function}
