\newacronym{HMM}{HMM}{hidden Markov model}
\newacronym{CNV}{CNV}{copy number variation}
\newacronym{LOH}{LOH}{loss of heterozygosity}
\newacronym{FDR}{FDR}{false-discovery rate}
\newacronym{BIC}{BIC}{Bayesian information criterion}
\newacronym{CPV}{CPV}{clinico-pathological variable}
\newacronym{SNV}{SNV}{single nucleotide variant}
\newacronym{LABQSR}{LA-BQSR}{local alignment and base quality score recalibration}
\newacronym{GATK}{GATK}{Genome analysis toolkit}
\newacronym{BAM}{BAM}{binary sequence alignment / map file}
\newacronym{indel}{indel}{insertion / deletion event}
\newacronym{BMC}{BMC}{Bayesian model comparison}
\newacronym{NGS}{NGS}{next-generation sequencing}
\newacronym{WGS}{WGS}{whole genome sequencing}
\newacronym{SNR}{SNR}{signal-to-noise ratio}
\newacronym{KDE}{KDE}{kernel density estimate}
\newacronym{MDS}{MDS}{multidimensional scaling}
\newacronym{IDAT}{IDAT}{Illumina data}
\newacronym{SIS}{SIS}{sure independence screening}
\newacronym{CPSS}{CPSS}{complementary pair subset selection}
\newacronym{FAST}{FAST}{feature aberration at survival times}
\newacronym{NMF}{NMF}{non-negative matrix factorization}
\newacronym{GEX}{GEX}{gene expression}
\newacronym{VST}{VST}{variance stabilizing transform}
\newacronym{GSVA}{GSVA}{gene set variation analysis}
\newacronym{SNMFL}{SNMF/L}{sparse non-negative matrix factorization, long variant}
\newacronym{MSigDB}{MSigDB}{molecular signatures database}
\newacronym{LASSO}{LASSO}{least absolute shrinkage and selection operator}
\newacronym{EMT}{EMT}{epithelial to mesenchymal transition}
\newacronym{NNLS}{NNLS}{non-negative least squares}
\newacronym{APGI}{APGI}{Australian Pancreatic Cancer Genome Initiative}
\newacronym{PDAC}{PDAC}{pancreatic ductal adenocarcinoma}
\newacronym{NCBI}{NCBI}{National Center for Biotechnology Information}
\newacronym{GEO}{GEO}{Gene Expression Omnibus}
\newacronym{BH}{BH}{Benjamini-Hochberg}
